crates/modelardb_manager/src/cluster.rs

```diff
@@ -113,6 +128,21 @@ impl Cluster {
         }
     }
 
+    /// Return the cloud node in the cluster that is currently most capable of running a query.
+    /// If there are no cloud nodes in the cluster, return [`ClusterError`](ModelarDbError::ClusterError).
+    pub fn query_node(&mut self) -> Result<Node, ModelarDbError> {
```

Comment:
I am not sure this is possible since the query node is owned by the function and the reference would break once the function returns. As mentioned above, nodes are extremely cheap so the simplest solution was chosen.

Reply:
It should not be a problem to return a reference to a query node stored in `query_queue` since that is stored in `self` and thus kept alive after the method call. However, if they are cheap to clone it should not be a problem and we can change this if profiling shows that the cloning is a problem.